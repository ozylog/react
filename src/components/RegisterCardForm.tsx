import React from 'react';
import styled from 'styled-components';

import { Form, Field } from 'react-final-form'

import { Title } from './../StyledComponents';

import { isRequired } from './../utils/validators';


const Container = styled.div`
  text-align: center;
`;

const FormContainer = styled.div`
  margin: 20px auto;
  max-width: 500px;

  input {
    margin-top: 10px;
    padding: 5px 7px;
  }

  button {
    margin-top: 10px;
    width: 100%;
    padding: 10px;
  }
`;

function RegisterCardForm(): React.ReactElement {

  const onSubmit = (values: any) => {
    console.log(values);
  };

  return (
    <Container>
      <Title>
        Welcome
      </Title>
      <Form
        onSubmit={onSubmit}
        render={({ handleSubmit, form, submitting, pristine, valid }) => (
          <form onSubmit={handleSubmit}>
            <FormContainer>
              <Field
                name='creditCard'
                component='input'
                type='text'
                placeholder='Credit card number'
                style={{ width: '100%'}}
                validate={isRequired}
              />

              <div style={{ display: 'flex' }}>
                <Field
                  name='cvc'
                  component='input'
                  type='password'
                  placeholder='CVC'
                  style={{ width: '50%'}}
                  validate={isRequired}
                />
                <Field
                  name='expiry'
                  component='input'
                  type='text'
                  placeholder='expiry'
                  style={{ width: '50%'}}
                  validate={isRequired}
                />
              </div>
              <button type="submit" disabled={submitting || pristine || !valid}>
                Submit
              </button>
            </FormContainer>
          </form>
        )}
      />
    </Container>
  );
}

export default React.memo(RegisterCardForm);
