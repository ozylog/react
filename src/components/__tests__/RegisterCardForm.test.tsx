import renderer from 'react-test-renderer';
import RegisterCardForm from '../RegisterCardForm';
import 'jest-styled-components';

describe('<RegisterCardForm />', () => {
  it('should render RegisterCardForm component', () => {
    const component = renderer.create(
      <RegisterCardForm />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
