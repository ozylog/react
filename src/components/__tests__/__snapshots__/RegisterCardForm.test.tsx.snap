// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterCardForm /> should render RegisterCardForm component 1`] = `
.c1 {
  font-size: 23px;
  font-weight: 500;
}

.c0 {
  text-align: center;
}

.c2 {
  margin: 20px auto;
  max-width: 500px;
}

.c2 input {
  margin-top: 10px;
  padding: 5px 7px;
  border: 1px solid #e5e5e5;
  border-radius: 3px;
  -webkit-appearance: none;
}

.c2 button {
  margin-top: 10px;
  width: 100%;
  padding: 10px;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Welcome
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="c2"
    >
      <input
        name="creditCard"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Credit card number"
        style={
          Object {
            "width": "100%",
          }
        }
        type="text"
        value=""
      />
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <input
          name="cvc"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="CVC"
          style={
            Object {
              "width": "50%",
            }
          }
          type="password"
          value=""
        />
        <input
          name="expiry"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="expiry"
          style={
            Object {
              "width": "50%",
            }
          }
          type="text"
          value=""
        />
      </div>
      <button
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
